.github/README.md

```diff
@@ -266,6 +266,81 @@ disabled in the current buffer. See the `god-exempt-predicates` variable and its
+
+[Evil][evil] is a popular Emacs package that provides modal editing in the style
+of Vi. While it doesn't always work well with God mode by default, there are
+some simple customizations that enable them to be used together smoothly.
+
+### Considerations
+
+When choosing a solution, consider the following questions:
+* Do I wish to use God mode for occasional, single commands (similar in spirit
+to `evil-execute-in-emacs-state`), or to use it in a sustained manner?
+* Do I need Evil-specific keybindings to be accessible within God mode, or will
+I use God mode to access default Emacs bindings? For example, when I press "v"
+in God mode, should it run `evil-visual-block` (Evil's default binding for C-v)
+or `scroll-up-command` (Emacs' default)?
+* Do I want God mode to have a dedicated state within Evil, or do I want to
+remain inside the classic Evil states while using it?
+
+### Using built-in functionality
+
+For running occasional, single commands via God mode, the built-in command
+`god-execute-with-current-bindings` works well with Evil without additional
+customization. All Evil bindings remain available when using this command,
+so e.g. pressing v will run `evil-visual-block`.
+
+For sustained usage of God mode, it gets trickier. Evil's keybindings in Motion,
+Normal, and Visual states generally override God mode. For example, if I am in
+Normal state and activate God mode, pressing v runs the Normal state binding for
+v (`evil-visual-char`) just as though God mode were not active. This behavior
+may be fine if all you want to do is run Emacs commands using God mode from
+within Insert state, since Evil doesn't bind most unmodified keys in Insert. If
+you need to access Evil bindings using God state, you'll need to apply the
+following customization for each Evil state `STATE` where you wish to do so:
+
+```emacs-lisp
+(evil-make-intercept-map
+     (evil-get-auxiliary-keymap god-local-mode-map 'god t t) 'STATE)
+```
+
+For example:
+
+```emacs-lisp
+(evil-make-intercept-map
+     (evil-get-auxiliary-keymap god-local-mode-map 'god t t) 'normal)
+```
+
+As a reminder these customizations are *not* necessary if you only use God mode
+via `god-execute-with-current-bindings`.
+
+### evil-god-state
+
+Another option is to use the package [evil-god-state][evil-god-state].
+This package provides a dedicated Evil state for using God mode.
+
+For running occasional, single commands via God mode, the package provides
+`evil-execute-in-god-state`. This works similar to
+`god-execute-with-current-bindings`. For sustained use, just use the command
+`evil-god-state`.
+
+An advantage of using a dedicated Evil state is that it provides built-in
+integration with the Evil mode line indicator. A disadvantage is that Evil's
+state-specific keybindings from other states are not available in God state.
+This is true whether using `evil-god-state` directly, or via
+`evil-execute-in-god-state`.  Thus, evil-god-state is most useful for accessing
```

Comment:
`evil-god-state` appears inconsistently as evil-god-state in certain places.
Mention `evil-god-state` (with backticks) in all places.